compiler/vm/src/builtin_functions.rs

```diff
@@ -266,9 +266,11 @@ impl Heap {
     }
     fn int_parse(&mut self, args: &[InlineObject]) -> BuiltinResult {
         unpack_and_later_drop!(self, args, |text: Text| {
-            let result = match BigInt::from_str(text.get()) {
-                Ok(int) => Ok(Int::create_from_bigint(self, true, int).into()),
-                Err(err) => Err(Text::create(self, true, &ToString::to_string(&err)).into()),
+            let result = if let Ok(int) = BigInt::from_str(text.get()) {
+                Ok(Int::create_from_bigint(self, true, int).into())
+            } else {
+                let not_an_integer = Text::create(self, true, "NotAnInteger");
+                Err(Tag::create_with_value_option(self, true, not_an_integer, None).into())
```

Comment:
```suggestion
                Err(Tag::create(self, not_an_integer).into())
```

However, I suggest adding the actual text we couldn't parse in there:

```suggestion
                Err(Tag::create_with_value(self, true, not_an_integer, text).into())
```

(Plus proper duping)